getsentry/sentry-java
# Integration testing for the sentry-native new unwinder

## :scroll: Description


Two UI tests:

- The first one spawns a service and crashes the app in the native layer.
- The second one checks if there is a file in the `.sentry-native` folder

Sentry Native Version:

- Per default, this uses `0.4.7` version of `sentry-native`. Please check out the correct version you want to test in the sub-module.

How to execute the tests on `AppCenter`:

Assuming you are running on a Mac.

- After creating the project in `AppCenter`, navigate to `test` and start creating a new test run.
- Select the devices.
<img width="1130" alt="Bildschirmfoto 2021-04-01 um 16 42 50" src="https://user-images.githubusercontent.com/3984453/113311718-e4f23780-9309-11eb-87c1-d0311e49a187.png">
- Select `Espresso` as the test framework.
<img width="1136" alt="Bildschirmfoto 2021-04-01 um 16 43 03" src="https://user-images.githubusercontent.com/3984453/113311740-e91e5500-9309-11eb-82bb-67796381eb83.png">
- Copy the `--app` and `--devices` parameters and finish.
<img width="1141" alt="Bildschirmfoto 2021-04-01 um 16 43 10" src="https://user-images.githubusercontent.com/3984453/113311848-005d4280-930a-11eb-9e5f-63a0d25f6a1c.png">

- Sign into `AppCenter` in the command line.
- `cd` into `sentry-samples/sentry-samples-android`
- Make sure you have a android device connected
- Make sure Android Studio is installed and check your build tools version in `/Library/Android/sdk/build-tools/`
- Create or add a keystore file named `sentry-samples-android-keystore`
- Run `appcenter` script with following parameters`./appcenter androidBuildToolsVersion keyAlias keyPass appCenterApp appCenterDevice` for example, this might look like `./appcenter 29.0.3 yourKeyAlias yourKeyPass denrase/Sentry-Android-Sample e669fa3b`
- Wait until `AppCenter` is finished on all devices...

Relates to #1359

## :bulb: Motivation and Context




## :green_heart: How did you test it?

- Locally and on Microsoft AppCenter

## :pencil: Checklist



## :crystal_ball: Next steps

- Depending on what the test results yield, we might do something with the envelope file and sent the to sentry.
- `AppCenter` also provides environmental variables, so we can also add info (device) to events.
